src/classes/ExpectObject_Tests.cls

```diff
@@ -0,0 +1,78 @@
+@isTest
+public class ExpectObject_Tests {
+    @isTest
+    static void ExpectEqualObjectsShouldNotThrow() {
+        Object one = new List<Account>{new Account(Name = 'Hello', Phone = '1')};
+        Object two = new List<Account>{new Account(Phone = '1', Name = 'Hello')};
+
+        Expect.that(one).shouldEqual(two);
```

Comment:
While testing strings this morning, I realized that!  I am going to update the PR in a few minutes, I've made a lot of progress today.